File: 09. Fragments/app/src/main/java/com/ralu/fragments/MainActivity.java
package com.ralu.fragments;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "Main";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onShowDetails(View view) {
        Log.d(TAG, "btn clicked");
        Intent intent = new Intent (this, DetailActivity.class);
        startActivity(intent);
    }
}

File: 10. FragmentComunication/app/src/main/java/com/example/headg/fragmentcomunication/MainActivity.kt
package com.example.headg.fragmentcomunication

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), RecyclerViewFragment.OnPersonSelectedListener {
private lateinit var recyclerViewFragment: RecyclerViewFragment
    private lateinit var itemFragment : ItemFragment
    private val RECYCLER_VIEW_FRAGMENT_TAG = "rvFragment"
    private val ITEM_FRAGMENT_TAG = "itemFragment"


    override fun onPersonSelected(person: Person) {
        val fragment: ItemFragment = supportFragmentManager.findFragmentByTag(ITEM_FRAGMENT_TAG) as ItemFragment
        fragment.personSelected(person)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    recyclerViewFragment = RecyclerViewFragment()
        itemFragment = ItemFragment()
        supportFragmentManager.beginTransaction()
                .add(R.id.rvFragmentContainer, recyclerViewFragment, RECYCLER_VIEW_FRAGMENT_TAG)
                .add(R.id.itemFragmentContainer,itemFragment, ITEM_FRAGMENT_TAG )
                .commit()
    }


}
